api/src/test/java/io/opentelemetry/trace/DefaultTracerTest.java

```diff
@@ -114,6 +114,22 @@ void testSpanContextPropagation_fromContext() {
     assertThat(span.getContext()).isSameAs(spanContext);
   }
 
+  @Test
+  void testSpanContextPropagation_fromContextAfterNoParent() {
```

Comment:
I believed this test would fail before this PR initially, but it's the other way round. Still adding this test to cover this case too.